### PR TITLE
Inductor cpp wrapper: fix codegen of ScatterFallback

### PR DESCRIPTION
**Pull request by chunyuan-w**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* #104575
* __->__ #104524

Fix cpp wrapper failure on TorchBench model `basic_gnn_edgecnn` and `hf_Reformer` which contain scatter OP.

cc @voznesenskym @penguinwu @EikanWang @jgong5 @Guobing-Chen @XiaobingSuper @zhuhaozhe @blzheng @Xia-Weiwen @wenzhe-nrv @jiayisunx @peterbell10 @ipiszy @ngimel @yf225 @chenyang78 @kadeng @muchulee8

### CODE REVIEW

File: test/inductor/test_cpp_wrapper.py [hunk 1]
```diff
@@ -179,6 +179,10 @@ class BaseTest(NamedTuple):
         BaseTest("test_relu"),  # multiple inputs
         BaseTest("test_repeat_interleave", "", test_cpu_repro.CPUReproTests()),
         BaseTest("test_scalar_input"),
+        BaseTest("test_scatter4"),
+        BaseTest("test_scatter5"),
+        BaseTest("test_scatter6"),
+        BaseTest("test_scatter_reduce2"),
```
Comment (jgong5) [inline test/inductor/test_cpp_wrapper.py:185]: Are we adding all `test_scatter` related tests?

**Review by jgong5 — APPROVED** [review]
LGTM. Seems we are not adding all `test_scatter` related UTs in cpp wrapper test? Can we do that?

File: test/inductor/test_cpp_wrapper.py (hunk 1 above)
Comment (chunyuan-w) [reply test/inductor/test_cpp_wrapper.py]: Scatter UTs have been added.

**Review by jgong5 — APPROVED** [review]

**Review by jansel — APPROVED** [review]

**Pull request closed without merging**